### PR TITLE
feat: add claim revocation

### PR DESCRIPTION
**Pull request by Harasz**
### Description

On-chain claim revocation features.

### Contributor checklist

- [x] Breaking changes - check for any existing interfaces changes that are not backward compatible, removed method etc.
- [x] Documentation - document your code, add comments for method, remember to check if auto generated docs were updated.
- [x] Tests - add new or updated existed test for changes you made.
- [x] Migration guide - add migration guide for every breaking change.
- [x] Configuration correctness - check that any configuration changes are correct ex. default URLs, chain ids, smart contract verification on [Volta explorer](https://volta-explorer.energyweb.org/) or [EWC explorer](https://explorer.energyweb.org/).

### CODE REVIEW

File: src/modules/claims/claims.service.ts [hunk 1]
```diff
@@ -113,6 +123,30 @@ export class ClaimsService {
   async init() {
     const chainId = this._signerService.chainId;
     this._claimManager = chainConfigs()[chainId].claimManagerAddress;
+
+    const signer = this._signerService.signer;
+    const provider = signer.provider;
+    const publicKey = await this._signerService.publicKey();
+    let ewSigner: EwSigner;
```
Comment (JGiter) [inline src/modules/claims/claims.service.ts:130]: why not export it from did-registry.service?

File: src/modules/claims/claims.service.ts [hunk 2]
```diff
@@ -843,6 +877,198 @@ export class ClaimsService {
...
+    return await this._claimRevocation.revokeClaim(namespace, subject, revoker);
+  }
+
+  /**
+   * Revoke On-Chain issued claims of the given namespace for multiple subjects. Namespace must be the same for all subjects.
+   *
+   * ```typescript
+   * claimsService.revokeMultipleClaim({
+   *     claims: [{
+   *         namespace: 'root.roles.energyweb.iam.ewc',
```
Comment (JGiter) [inline src/modules/claims/claims.service.ts:918]: Why not move namespace from array elements to root of claims type?

File: src/modules/claims/claims.service.ts [hunk 3]
```diff
@@ -843,6 +877,198 @@ export class ClaimsService {
...
+    ];
+    let namespace = claims[0]?.namespace;
+
+    if (claimIds) {
+      const claimsDetails = await Promise.all([
+        ...claimIds.map(async (claimId) => {
+          return await this.getClaimById(claimId);
+        }),
+      ]);
+      claimsDetailsToRevoke = [
```
Comment (JGiter) [inline src/modules/claims/claims.service.ts:958]: Is this means that claims should not be provided along with claimId's? If so we need to mention this in description or better exclude such possibility at all

File: src/modules/claims/claims.service.ts (hunk 2 above)
Comment (Harasz) [reply src/modules/claims/claims.service.ts:918]: The idea behind that is to make interface simple, so you only provide `claims` or `claimIds`.

File: src/modules/claims/claims.service.ts (hunk 3 above)
Comment (Harasz) [reply src/modules/claims/claims.service.ts:958]: Yes, for all revocation method you can specify claim details or just simply pass claim id. And claim id will always override claim details. I've added comments to documentation and example uses

File: src/modules/claims/claims.service.ts (hunk 1 above)
Comment (Harasz) [reply src/modules/claims/claims.service.ts:130]: Done

**Review by JGiter — APPROVED** [review]

**Review by JGiter — APPROVED** [review]

**Pull request merged**